SharedSource/UnitTests/Reflection/TypeExtensionsTest.cs

```diff
@@ -0,0 +1,170 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+
+using System;
+using System.Reflection;
+using NUnit.Framework;
+using Remotion.Reflection;
+
+#nullable enable
+// ReSharper disable once CheckNamespace
+namespace Remotion.UnitTests.Reflection
+{
+  [TestFixture]
+  public class TypeExtensionsTest
+  {
+    [Test]
+    public void GetAssemblyQualifiedNameSafe_SystemType_AssemblyQualifiedName ()
```

Comment:
Namings:
GetAssemblyQualifiedNameSafe_**With**SystemType_**Returns**AssemblyQualifiedName

Reply:
I can see that and it's probably a matter of style. re-motion does employ the context words in the test fixtures we've written since xUnit Patterns has been published so the mental expectation does exist. So there's the mental expectation to see them and especially the Returns/Throws information is very important since it helps differentiate if the test name is simply done slopily (skipping the Returns information) or this actually being the returned value.